service/grails-app/services/org/olf/licenses/LicenseHousekeepingService.groovy

```diff
@@ -0,0 +1,68 @@
+package org.olf.licenses
+
+import grails.gorm.transactions.Transactional
+import org.olf.licenses.License
+import org.olf.licenses.LicenseAmendment
+import com.k_int.web.toolkit.refdata.RefdataValue
+
+/**
+ * This service works at the module level, it's often called without a tenant context.
+ */
+@Transactional
+public class LicenseHousekeepingService {
+
+  public void triggerHousekeeping() {
+    this.checkUnsetValues();
+  }
+
+  private List<List<String>> batchFetchAmendments(final int platformBatchSize, int platformBatchCount) {
+    List<List<String>> amendments = LicenseAmendment.createCriteria().list ([max: platformBatchSize, offset: platformBatchSize * platformBatchCount]) {
+      order 'id'
```

Comment:
We want this criteria to find the amendments where endDateSemantics are null, ie we need a block in this criteria that does something like:
```
isNull('endDateSemantics')
```

When you add this, make sure you're checking that the batch fetching is only grabbing those licenses/amendments that have endDateSemantics null to check your criteria does what we expect. We don't want to run through this for every license/amendment

Reply:
https://docs.grails.org/4.0.11/ref/Domain%20Classes/createCriteria.html

Look through this page for the `isNull` constraint